### PR TITLE
Add the syntax highlighting.

### PR DESCRIPTION
**Pull request by ThreeMonth03**
According the the discussion in issue [#395](https://github.com/solvcon/modmesh/discussions/395), I've completed adding syntax highlighting to all the code blocks in [STYLE.rst](https://github.com/solvcon/modmesh/blob/5cc3cc326ed66b72315f7116ee9f1a34584d69eb/STYLE.rst).

### CODE REVIEW

File: STYLE.rst [hunk 1]
```diff
@@ -15,7 +15,7 @@ style when adding new code and changing existing code. The rules of thumb are:
 1. The linters must be clean. Before creating and updating a
    `pull request <https://docs.github.com/en/pull-requests/>`__, run:
 
-   .. code-block::
+   .. code-block:: bash
```
Comment (ThreeMonth03) [inline STYLE.rst:18]: Bash

File: STYLE.rst [hunk 2]
```diff
@@ -259,7 +259,7 @@ The inclusion guard uses ``#pragma once`` in the first line before everything.
 Always use path-first inclusion (angle branket). Do not use current-first
 (double quote).
 
-.. code-block::
+.. code-block:: cpp
```
Comment (ThreeMonth03) [inline STYLE.rst:262]: cpp

File: STYLE.rst [hunk 3]
```diff
@@ -448,7 +448,7 @@ When creating a new file, put the following text at the top of the file
 (replace ``<Year>`` with the year you create the file and ``<Your Name>`` with
 your name and maybe email).  The license text formatted for C++ files:
 
-.. code-block::
+.. code-block:: cpp
```
Comment (ThreeMonth03) [inline STYLE.rst:451]: cpp

File: STYLE.rst [hunk 4]
```diff
@@ -480,7 +480,7 @@ your name and maybe email).  The license text formatted for C++ files:
 
 The license text formatted for Python files:
 
-.. code-block::
+.. code-block:: python
```
Comment (ThreeMonth03) [inline STYLE.rst:483]: python

**Review by ThreeMonth03 — COMMENTED** [review]
In this PR, I added syntax highlighting to all the code blocks. This is a minor change for the documentation.

**Comment by ThreeMonth03** [comment]
> In this PR, I added syntax highlighting to all the code blocks. This is a minor change for the documentation.

@yungyuc When you are available, you could review this pr. Thank you very much.

**Review by tigercosmos — APPROVED** [review]
LGTM

**Comment by yungyuc** [comment]
Thanks for the fix.

**Pull request merged**